src/main/java/com/hivemq/mqtt/topic/SubscriberWithIdentifiers.java

```diff
@@ -56,7 +56,7 @@ public SubscriberWithIdentifiers(final @NotNull String subscriber, final int qos
         this.qos = qos;
         this.flags = flags;
         this.sharedName = sharedName;
-        this.subscriptionIdentifier = ImmutableIntArray.of();
+        this.subscriptionIdentifier = ImmutableIntArray.of(subscriptionIdentifier);
```

Comment:
```suggestion
        this.subscriptionIdentifier = (subscriptionIdentifier == null) ? ImmutableIntArray.of():ImmutableIntArray.copyOf(subscriptionIdentifier);
```

Reply:
Thanks for your contribution. 
When creating an ImmutableIntArray from a Collection another signature must be used.
This constructor is only used in a few unit tests that do not test subscription identifiers as such.

Reply:
Yes, your suggestion makes sense.

However, given the `NotNull` annotations, should not the **nullability** be check by `checkNotNull`?